### PR TITLE
Remove special character validations for names (patients and users)

### PR DESCRIPTION
**Pull request by rithviknishad**
### Proposed Changes
- This reverts commit ff63195d38f5bdfb684c2f6c429cffde217063ec. Reason: https://github.com/ohcnetwork/care_fe/issues/9143#issuecomment-2491409154
- Also removes numeric validation check for care user's names.

## Summary by CodeRabbit

- **New Features**
	- Introduced a minimum character length requirement for name fields in both user and patient registration forms.

- **Bug Fixes**
	- Ensured pincode validation is retained and correctly implemented in the Patient Register component.
	- Improved error messaging for name fields to provide clearer feedback on validation requirements.

- **Refactor**
	- Updated form validation logic across components to enhance clarity and maintainability while preserving functionality for other fields.
	- Simplified validation process for names to check for a minimum length of three characters.

### CODE REVIEW

File: src/common/validation.tsx [hunk 1]
```diff
@@ -37,11 +37,6 @@ export const getRandomNumbers = (min: number, max: number) => {
   return Math.floor(Math.random() * max) + min;
 };
 
-export const validateName = (name: string) => {
```
Comment (bodhish) [inline src/common/validation.tsx:40]: I think we should have a min length validation?

**Review by bodhish — CHANGES_REQUESTED** [review]

File: src/common/validation.tsx (hunk 1 above)
Comment (rithviknishad) [reply src/common/validation.tsx:40]: done

**Comment by nihal467** [comment]
LGTM

**Review by bodhish — APPROVED** [review]
LGTM

**Pull request merged**